openitg/openitg
# Single MSVC (updated) Project with tomcrypt and io additions

Since I did not get any feedback on #99 I decided to go ahead and do this. Windows compilation is EASY now.

Addresses issues #25, #66 (closed but unaddressed) and #99 

Prereqs: Install the DX9 SDK.

Procedure:
Open the stepmania-net2010.sln in any version of visual studio (2010 and newer; only 2010, 2013 and 2015 tested) and with VERY minor changes, your good to go!

MSVC compiler selection:
By default the project is configured for msvc2015 with xp support. To change this, in the solution explorer, simply right click each of the 5 projects and under Configuration Properties -> General look for the option of "Platform Toolset" You can set this to any toolchain you have installed. (To save clicks, make sure you select the All configurations configuration option in the upper left of each project selected). If you need a platform that I haven't prebuilt the libs for, look in the extern folder and drop your compiled lib in there.

Note that MSVC 2013 has a bug, it required use of the /FS switch (Or you have to build the project twice), however, MSVC 2010 does NOT support this switch and the switch is ONLY NEEDED by MSVC 2013. If you run an older version of MSVC, for each project, open the properties and (To save clicks, make sure you select the All configurations configuration option in the upper left of each project selected) simply go to Configuration Properties -> C/C++ -> Command line and delete the /FS flag from each of the 5 projects.

Build Instructions:
*To build, right click on the solution and say "Clean". Then right click on OpenITG and select Rebuild.*
That's it!


TO TEST:
Linux Compilation - Tomcrypt/tommath was updated and shoved into the project for testing purposes, the two are sadly intertwined because it's such a deep dependency. I do not anticipate compilation problems on linux but the performance improvements are drastic with this update and well worth the look. I also moved some folders around but they should be specific to windows - to make dependent libs easier to manage. I cannot test Linux compiles so I'd appreciate if someone could.

XBOX Compilation:
I cannot source a copy of msvc++2003 to test the xbox project and update it. You should just need to point it at some new directories in the solution is all though if theres a problem.
